src/lib/persistor.ts

```diff
@@ -1,13 +1,26 @@
+import { MoodleApi } from '../moodleapi'
+import { getCurrentContext } from './utils'
+
 export interface Persistor {
   get: (key: string) => Promise<string | null>
   put: (key: string, value: string) => Promise<void>
 }
 
-export const browserPersistor: Persistor = {
-  get: async (key: string): Promise<string | null> => {
-    return localStorage.getItem(key)
-  },
-  put: async (key: string, value: string) => {
-    localStorage.setItem(key, value)
+export const getMoodlePersistor = (): Persistor | undefined => {
+  const { courseId } = getCurrentContext()
+
+  if (courseId === undefined || window.M === undefined) {
```

Comment:
We can add the following to use the `ENV` list to enforce disabling:

```javascript
  if (courseId === undefined || window.M === undefined || host === undefined || !ENV.OS_RAISE_ENABLE_MOODLE_PERSISTOR_HOSTS.includes(host)) {
    return undefined
  }
```

The `host` used :point_up: comes from the context.